### PR TITLE
puller(ticdc): refactor ddl puller remove verbose code and reduce logs

### LINKED ISSUE #10296: cdc refactor ddl puller to make log more clear and simplify code
as title

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #10296

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: cdc/entry/mounter.go [hunk 1]
```diff
@@ -316,22 +316,28 @@ func ParseDDLJob(tblInfo *model.TableInfo, rawKV *model.RawKVEntry, id int64) (*
...
 func parseJob(v []byte, startTs, CRTs uint64) (*timodel.Job, error) {
-	job := &timodel.Job{}
-	err := json.Unmarshal(v, job)
+	var job timodel.Job
+	err := json.Unmarshal(v, &job)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	log.Debug("get new DDL job", zap.String("detail", job.String()))
+	// TiDB write DDL Binlog for every DDL Job,
```
Comment (asddongmen) [inline cdc/entry/mounter.go:324]:
```suggestion
	One DDL has multiple states from start to finish, but TiCDC only listens to and processes Job with the "Done" status. Other Job statuses should be ignored.
```

File: cdc/entry/mounter.go [hunk 2]
```diff
@@ -316,22 +316,28 @@ func ParseDDLJob(tblInfo *model.TableInfo, rawKV *model.RawKVEntry, id int64) (*
...
+	// we must ignore jobs that are cancelled or rollback
+	// For older version TiDB, it writes DDL Binlog in the txn
+	// that the state of job is changed to *synced*
+	// Now, it writes DDL Binlog in the txn that the state of
+	// job is changed to *done* (before change to *synced*)
+	// At state *done*, it will be always and only changed to *synced*.
```
Comment (asddongmen) [inline cdc/entry/mounter.go:330]: These comments here is a bit confusing for me to read.

File: cdc/processor/processor.go [hunk 3]
```diff
@@ -1007,10 +1007,6 @@ func (d *ddlHandler) Run(ctx context.Context, _ ...chan<- error) error {
 			if jobEntry.OpType == model.OpTypeResolved {
 				d.schemaStorage.AdvanceResolvedTs(jobEntry.CRTs)
 			}
-			err := jobEntry.Err
```
Comment (asddongmen) [inline cdc/processor/processor.go:1010]: why remove this?

File: cdc/puller/ddl_puller.go [hunk 4]
```diff
@@ -115,36 +115,35 @@ func (p *ddlJobPullerImpl) handleRawKVEntry(ctx context.Context, ddlRawKV *model
...
 	case <-ctx.Done():
 		return ctx.Err()
 	case p.outputCh <- jobEntry:
 	}
+
+	log.Info("one new ddl job is put into output channel",
+		zap.String("namespace", p.changefeedID.Namespace),
+		zap.String("changefeed", p.changefeedID.ID),
+		zap.String("query", job.Query),
+		zap.Uint64("startTs", job.StartTS))
```
Comment (asddongmen) [inline cdc/puller/ddl_puller.go:146]: Please also print it's FinishedTs

File: cdc/puller/ddl_puller.go [hunk 5]
```diff
@@ -391,8 +390,7 @@ func (p *ddlJobPullerImpl) handleJob(job *timodel.Job) (skip bool, err error) {
 				zap.String("changefeed", p.changefeedID.ID),
 				zap.String("schema", job.SchemaName),
 				zap.String("table", job.TableName),
-				zap.String("query", job.Query),
-				zap.Stringer("job", job))
+				zap.String("query", job.Query))
```
Comment (asddongmen) [inline cdc/puller/ddl_puller.go:393]: ditto

File: cdc/puller/ddl_puller.go [hunk 6]
```diff
@@ -401,7 +399,6 @@ func (p *ddlJobPullerImpl) handleJob(job *timodel.Job) (skip bool, err error) {
 				zap.String("schema", job.SchemaName),
 				zap.String("table", job.TableName),
 				zap.String("query", job.Query),
```
Comment (asddongmen) [inline cdc/puller/ddl_puller.go:401]: ditto

File: cdc/puller/ddl_puller.go [hunk 7]
```diff
@@ -410,26 +407,24 @@ func (p *ddlJobPullerImpl) handleJob(job *timodel.Job) (skip bool, err error) {
...
 			"discard the ddl job",
-			zap.Uint64("jobFinishedTS", job.BinlogInfo.FinishedTS),
-			zap.Uint64("pullerResolvedTs", p.getResolvedTs()),
 			zap.String("namespace", p.changefeedID.Namespace),
 			zap.String("changefeed", p.changefeedID.ID),
 			zap.String("schema", job.SchemaName),
 			zap.String("table", job.TableName),
-			zap.String("query", job.Query),
-			zap.String("job", job.String()))
+			zap.Uint64("jobFinishedTS", job.BinlogInfo.FinishedTS),
```
Comment (asddongmen) [inline cdc/puller/ddl_puller.go:414]: Please also log it StartTs

File: cdc/puller/ddl_puller.go [hunk 8]
```diff
@@ -680,48 +658,31 @@ func NewDDLPuller(ctx context.Context,
...
-			h.lastResolvedTsAdvancedTime = h.clock.Now()
-			atomic.StoreUint64(&h.resolvedTS, jobEntry.CRTs)
-		}
-		return nil
-	}
-	job, err := jobEntry.Job, jobEntry.Err
-	if err != nil {
-		return errors.Trace(err)
-	}
+func (h *ddlPullerImpl) add2Pending(job *timodel.Job) {
```
Comment (asddongmen) [inline cdc/puller/ddl_puller.go:665]:
```suggestion
func (h *ddlPullerImpl) addToPending(job *timodel.Job) {
```

File: cdc/entry/mounter.go (hunk 2 above)
Comment (3AceShowHand) [reply cdc/entry/mounter.go:330]: it's from other code snippet.

File: cdc/processor/processor.go (hunk 3 above)
Comment (3AceShowHand) [reply cdc/processor/processor.go:1010]: I think this error won't be checked, because it's already checked when the error happens.

File: cdc/puller/ddl_puller.go [hunk 9]
```diff
@@ -442,20 +415,46 @@ func (p *ddlJobPullerImpl) handleJob(job *timodel.Job) (skip bool, err error) {
 		return true, errors.Trace(err)
 	}
 
+	defer func() {
```
Comment (asddongmen) [inline cdc/puller/ddl_puller.go:418]: Why did you move this defer block here? It can't cover the situation that would happen at line 413.

**Comment by asddongmen** [comment]
Please make sure that the content of a `DDL Job` is printed at lease once in a log, such as in the `run` method in `ddl_puller.go`. This will be very helpful in debugging.

**Review by asddongmen — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

File: cdc/entry/schema_storage.go [hunk 10]
```diff
@@ -74,8 +74,8 @@ func NewSchemaStorage(
 ) (SchemaStorage, error) {
 	var (
 		snap    *schema.Snapshot
-		err     error
 		version int64
+		err     error
```
Comment (CharlesCheung96) [inline cdc/entry/schema_storage.go:78]:
```suggestion
		err     error
```

Comment (3AceShowHand) [reply cdc/entry/schema_storage.go:78]: This cannot be removed, otherwise the `snap` become not used, report by the linter.
